physical/postgresql.go

```diff
@@ -0,0 +1,268 @@
+}
+
+func (b *PostGresBackend) Delete(k string) error {
+	Trace.Println("> Delete")
+	b.l.Lock()
+	defer b.l.Unlock()
+
+	db, err := sql.Open("postgres", b.Url)
+	if err != nil {
+		Trace.Println("< Delete error db")
+		Error.Fatal(err)
+	}
+	defer db.Close()
+
+	txn, err := db.Begin()
+	if err != nil {
+		Trace.Println("< Delete error begin")
+		Trace.Fatal(err)
+	}
+
+	stmt, err := txn.Prepare("delete from vault where key = $1")
+	if err != nil {
+		Trace.Println("< Delete error prepare")
+		Error.Fatal(err)
+	}
+
+	_, err = stmt.Exec(k)
+	if err != nil {
+		Trace.Println("< Delete error exec")
+		Error.Fatal(err)
+	}
+
+	err = stmt.Close()
+	if err != nil {
+		Trace.Println("< Delete error close")
+		Error.Fatal(err)
+	}
+
+	err = txn.Commit()
+	if err != nil {
+		Trace.Println("< Delete error commit")
+		Error.Fatal(err)
+	}
+
+	Trace.Println("< Delete")
+	return err
+}
+
+func (b *PostGresBackend) Get(k string) (*Entry, error) {
+	Trace.Println("> Get")
+	b.l.Lock()
+	defer b.l.Unlock()
+
+	db, err := sql.Open("postgres", b.Url)
+	if err != nil {
+		Trace.Println("< Get error db")
+		Error.Fatal(err)
+	}
+	defer db.Close()
+
+	stmt, err := db.Prepare("select value from vault where key = $1")
+	defer stmt.Close()
+	if err != nil {
+		Trace.Println("< Get error select")
+		Error.Fatal(err)
+	}
+
+	var value []byte
+	row, err := stmt.Query(k)
+	if err != nil {
+		Error.Println(err)
+		Trace.Println("< Get error query")
+		return nil, err
+	}
+
+	if row.Next() {
+		row.Scan(&value)
+		entry := Entry{k, value}
+
+		return &entry, nil
+	}
+	Trace.Println("< Get")
+	return nil, err
+
+}
+
+func (b *PostGresBackend) Put(entry *Entry) error {
+	Trace.Println("> Put")
+	b.l.Lock()
+	defer b.l.Unlock()
+
+	db, err := sql.Open("postgres", b.Url)
+	if err != nil {
+		Trace.Println("< Put error db")
+		Error.Fatal(err)
+	}
+	defer db.Close()
+
+	rows, err := db.Query("select key from vault where key = $1", entry.Key)
+	if err != nil {
+		Trace.Println("< Put error select")
+		Error.Fatal(err)
+	}
+	defer rows.Close()
+
+	txn, err := db.Begin()
+	if err != nil {
+		Trace.Println("< Put error tran begin")
+		Error.Fatal(err)
+	}
+	next := rows.Next()
+	Trace.Printf("D Put - is there next row? %t", next)
+
+	// need to update if already there
+	if next {
+		stmt, err := txn.Prepare("update vault set value =  $1, updated_at = $2 where key = $3")
+		if err != nil {
+			Trace.Println("< Put error update")
+			Error.Fatal(err)
+		}
+
+		time := time.Now()
+		_, err = stmt.Exec(entry.Value, time, entry.Key)
+		if err != nil {
+			Trace.Println("< Put error exec")
+			Error.Fatal(err)
+		}
+
+		err = stmt.Close()
+		if err != nil {
+			Trace.Println("< Put error close")
+			Error.Fatal(err)
+		}
+
+	} else {
+		stmt, err := txn.Prepare("insert into vault (key, value, created_at, updated_at) values ($1, $2, $3, $4)")
+		if err != nil {
+			Trace.Println("< Put error insert prepare")
+			Error.Fatal(err)
+		}
+
+		time := time.Now()
```

Comment:
It might be worth using According to the [package docs](http://golang.org/pkg/time/#Now), `Now()` returns the machine's local time. Given that the audit log [uses UTC](https://github.com/hashicorp/vault/blob/e2e077964b5200208a7554963d49f55ec98ade57/audit/format_json.go#L33) it might be worthing using it here too. It seems it's even possible to have postgres [use the current UTC time for the column's default value](http://stackoverflow.com/a/16610360/149297).